airflow/ti_deps/deps/prev_dagrun_dep.py

```diff
@@ -66,6 +67,23 @@ def _get_dep_statuses(self, ti, session, dep_context):
 
         previous_ti = last_dagrun.get_task_instance(ti.task_id, session=session)
         if not previous_ti:
+            if ti.task.ignore_first_depends_on_past:
+                historical_ti = (
+                    session.query(TI)
+                    .filter(
+                        TI.dag_id == ti.dag_id,
+                        TI.task_id == ti.task_id,
+                        TI.execution_date < ti.execution_date,
+                    )
+                    .first()
+                )
```

Comment:
```suggestion
                historical_ti = (
                    session.query(literal(True))
                    .filter(
                        TI.dag_id == ti.dag_id,
                        TI.task_id == ti.task_id,
                        TI.execution_date < ti.execution_date,
                    )
                    .first()
                )
```

This would save some deserialisation effort since we don’t actually care about the selected task instance, but only whether it exists.

Reply:
Alternatively we could do `session.query(func.count()).filter(...).scalar() > 0`, which is probably slower, but still reasonable since the filtered columns are indexed, and arguably most readable?

Reply:
good idea. I have tried to use `exist` but some db does not support that. let me update it